corehq/messaging/scheduling/static/scheduling/js/conditional_alert_main.js

```diff
@@ -1,36 +1,81 @@
-
 hqDefine("scheduling/js/conditional_alert_main", [
+    'jquery',
+    'knockout',
+    'hqwebapp/js/initial_page_data',
     'data_interfaces/js/case_rule_criteria',
     'scheduling/js/create_schedule.ko',
     'data_interfaces/js/make_read_only',
-], function () {
-    $("#basic_continue").on('click', function () {
-        if ($("#id_conditional-alert-name").val().trim()) {
-            // remove error message
-            $("#required_error_message").addClass('hidden');
-            // move to next tab
-            $("#rule-nav").removeClass('hidden');
-            $("a[href='#rule']").trigger('click');
-        } else {
-            // show error message
+], function ($, ko, initialPageData) {
```

Comment:
This is using knockout now but not really the way knockout is meant to be used. Instead of directly manipulating DOM elements, knockout should update the model's state, and then HTML elements use bindings to reflect that state. In general, if you're in knockout code and are using `$(...)` to find elements, something fishy is going on.

So for the basic tab, instead of calling `handleBasicNavContinue` when the user clicks continue, then running validation and if it fails, calling `showRequiredErrorMessage` to search for `required_error_message` and remove the `hidden` class, you'd:
- Add an observable to the knockout model to keep track of the rule's name: `self.name = ko.observable();` in the model and then `data-bind='value: name'` on the name input.
- Change `basicTabValid` to a computed observable that checks the value of that observable: `self.basicTabValid = ko.computed(function () { return !_.isEmpty(self.name().trim()); })`
- Add an `enable: basicTabValid` binding to the continue button. Remove the explicit validation in `handleBasicNavContinue`. This'll keep the continue button disabled until the instant a name is entered.
- Because this is just a required field, I'd remove the error message altogether. If there's an asterisk next to "Name" and the continue button is disabled when name is blank, that's enough feedback. But if this were more complex validation, I'd handle it by adding a binding like `visible: !basicTabValid()` to the error message element.